src/main/java/org/neo4j/sql2cypher/Translator.java

```diff
@@ -461,15 +448,55 @@ else if (c instanceof QOM.IsNotNull e) {
 		}
 	}
 
-	private Node node(Table<?> t) {
+	private PatternElement resolveTableOrJoin(Table<?> t) {
+		if (t instanceof QOM.Join<?> ta && ta.$on() instanceof QOM.Eq<?> eq) {
```

Comment:
@lukaseder is this the best way to determine a simple equals join?

Reply:
yeah, perhaps we can add some failing tests, then for option 1, we can move the predicates to the WHERE clause.
NOT DISTINCT FROM - need to figure out how to express that in cypher.

Multi-key pair equalities should work the same as single ones (just map to multiple rels), would be probably good to filter for equalities, process them all and add the remaining predicates to the WHERE.

In terms of rare syntax we can just fail with "not-implemented" for the time being.